pants.toml

```diff
@@ -116,10 +111,17 @@ ignore_adding_targets = [
 venv_use_symlinks = true
 
 [python]
-experimental_lockfile = "3rdparty/python/lockfiles/user_reqs.txt"
 interpreter_constraints = [">=3.7,<3.10"]
 macos_big_sur_compatibility = true
 
+# To use resolves, switch which line is commented out.
+experimental_lockfile = "3rdparty/python/lockfiles/user_reqs.txt"
+#enable_resolves = true
+
+[python.experimental_resolves]
+python-default = "3rdparty/python/lockfiles/user_reqs.lock"
+python-testprojects = "testprojects/3rdparty/python/user_reqs.lock"
```

Comment:
@stuhood question about the theory of multiple resolves? Would it make sense for our `build-support` code to be a separate resolve? This would I think resolve the issue we've had in the past of pantsbuild.pants code accidentally using build-support requirements, thanks to the visibility change we made recently.

Reply:
> Would it make sense for our `build-support` code to be a separate resolve? This would I think resolve the issue we've had in the past of pantsbuild.pants code accidentally using build-support requirements, thanks to the visibility change we made recently.

Only if it needs to be. The theory behind separate resolves is that having a single resolve of user code is better for compatibility, because it eases sharing of code. It also reduces the number of permutations and unique requirements that are in use: in the presence of range requirements, it's possible for each different resolve to end up with a different item from the range. when all requirements are considered together, the total number of permutations is reduced, and one version is chosen (again, if it can be), which should improve performance and the amount downloaded.

So: you shouldn't create more resolves unless you have to because they contain incompatible requirements.